pkg/etcd/etcd.go

```diff
@@ -171,7 +171,11 @@ func (e *ETCD) Reset(ctx context.Context, clientAccessInfo *clientaccess.Info) e
 	if err := e.setName(true); err != nil {
 		return err
 	}
-
+	// touch a file to avoid multiple resets
+	resetFile := filepath.Join(e.config.DataDir, "db", "reset-flag")
```

Comment:
you filepath.Join this same set of strings twice, it probably deserves a function alongside the other ones so that it's all in one place.

https://github.com/rancher/k3s/blob/master/pkg/etcd/etcd.go#L108